src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -264,18 +265,29 @@ public void SetAll(bool value)
         ** Exceptions: ArgumentException if value == null or
         **             value.Length != this.Length.
         =========================================================================*/
-        public BitArray And(BitArray value)
+        public unsafe BitArray And(BitArray value)
         {
             if (value == null)
                 throw new ArgumentNullException(nameof(value));
             if (Length != value.Length)
                 throw new ArgumentException(SR.Arg_ArrayLengthsDiffer);
 
-            for (int i = 0; i < m_array.Length; i++)
+            int i = 0;
+            if (Sse2.IsSupported && m_array.Length >= 4)
```

Comment:
Do we have any metrics for the average length of a `BitArray`?

Reply:
I tried to gain some insights here that might be relevant: https://github.com/dotnet/corefx/pull/33367#discussion_r234039122